stable/jenkins/templates/config.yaml

```diff
@@ -51,7 +51,12 @@ data:
               <name>{{ .Values.agent.podName }}</name>
               <instanceCap>2147483647</instanceCap>
               <idleMinutes>{{ .Values.agent.idleMinutes }}</idleMinutes>
-              <label>{{ .Release.Name }}-{{ .Values.agent.componentName }} {{ .Values.agent.customJenkinsLabels  | join " " }}</label>
+              {{- $additionalLabels := "" }}
+              {{- if .Values.agent.customJenkinsLabels }}
+              {{- $tmp := join " " .Values.agent.customJenkinsLabels }}
+              {{- $additionalLabels = printf " %s" $tmp }}
+              {{- end  }}
+              <label>{{ .Release.Name }}-{{ .Values.agent.componentName }}{{ $additionalLabels }}</label>
```

Comment:
What do you want to achieve here? My feeling is that it makes the code harder to read.

Reply:
@torstenwalter I think @sami-alajrami is refering to this code in his first post

```
The default config.xml template leaves a space after the default PodLabel if custom labels are not provided. The kubernetes plugin replaces spaces with an x character making the network policy podSelector fail to match the label on the agent.
```

If custom labels are not set, the label will have an extra space at the end, which the kubernetes plugin will replace with an X as spaces are not allowed in kubernetes labels. So the chart needs to check if custom labels are used, and only join them with a space if so. Network policies will not find the labels, if the Kubernetes plugin adds a X because of a space made by the chart.

Reply:
What I try to achieve is to prevent adding an unnecessary whitespace after the default label and only add it if there are `customJenkinsLabels`. This is because the Jenkins k8s plugin replaces the whitespace with `x` for some reason and therefore the label no longer matches the expected label on the network policy.

A  (possibly) simpler alternative would be :
```
{{- $tmp := join " " .Values.agent.customJenkinsLabels }}
{{- $labels := printf "%s-%s %s" .Release.Name .Values.agent.componentName $tmp }}
<label>{{ $labels | trim  }}</label>
```

Reply:
I have pushed the simplified code for trimming trailing whitespaces in the labels passed in the config.xml. @torstenwalter can I mark this as resolved now? or you still have reservations?